neurots/generate/tree.py

```diff
@@ -284,13 +284,14 @@ def next_point(self):
                         child.first_point()
                     self.active_sections.remove(section_grower)
 
+                    # we need this so that the pathlenght match due the child.first_point() in bifs
+                    for section in self.active_sections:
```

Comment:
yes, basically the growth happens uniformly, as we loop over all active sections, and and a point, then repeat. So at a given time,all the tips of growing sections are at the same path distance from soma. With the child.first_point() above, if we bifurcate (think of a small collateral off a main branch), then we will add one extra point on the main branch wrt to all the others which do not bifurcate at this path distance. This will result of the bifucrated branch being one step ahead in path distance wrt to the others. In certain edge cases (like for an early bifurcation in apical to have 2 apical sub-trunks), as the choice of bifurcating is proportional to path distance, this will induce a bias (in the case of bi-trunk, the first truck to bifurcate with an oblique will be ahead, and get all the obliques, leaving the other sub-trunk without branches, which is not correct). We thus need to ensure that the growth is completely uniform in path length.

Reply:
Ok, I see! Thanks for the details.
The general principle looks good to me, so I let you fix the tests :wink: